src/mustgather.rs

```diff
@@ -23,7 +23,11 @@ pub struct MustGather {
 
 impl MustGather {
     /// Build a MustGather from a path to a directory containing the root.
-    pub fn from(path: String) -> Result<MustGather> {
+    pub fn from(path: impl AsRef<Path>) -> Result<MustGather> {
```

Comment:
Note here that accepting an `impl AsRef<Path>` means that this function can directly accept a `String`, `&str`, `PathBuf`, `&Path` etc.

However using generics would then duplicate the text, so we dispatch all the heavy lifting to a helper function which just takes `&Path`.

Reply:
very cool!